common/src/main/java/com/wynntils/models/marker/type/MarkerProvider.java

```diff
@@ -4,10 +4,13 @@
  */
 package com.wynntils.models.marker.type;
 
+import com.wynntils.services.map.pois.Poi;
 import java.util.stream.Stream;
 
-public interface MarkerProvider {
+public interface MarkerProvider<T extends Poi> {
     Stream<MarkerInfo> getMarkerInfos();
 
+    Stream<T> getPois();
```

Comment:
This partially overlaps getMarkerInfos().map(m -> m.locationSupplier.get()) ... But I think we can keep doing it this way and hope that I get to finish the mapdata rewrite soon and then we'll have the tools do to this better.

Reply:
Yes, the poi system wasn't really ready for this. But we need to show quest/lootrun waypoints on the map..